File: Codeforces/Round 590 Div. 3/C__AC_bLuck_.py
import sys

def main():
  sys.stdin = open('in.txt', 'r')
  t = int(sys.stdin.readline())
  for _ in range(t):
    n = int(sys.stdin.readline())
    a = list(sys.stdin.readline())
    b = list(sys.stdin.readline())
    
    chk = False #1st row
    for i in range(n):
      
      if chk is False:
        if int(a[i]) > 2:
          if int(b[i]) <= 2:
            chk = False
            break
          chk = not (chk)
      else:
        if int(b[i]) > 2:
          if int(a[i]) <= 2:
            chk = False
            break
          chk = not (chk)
    print("YES") if chk is True else print("NO")

if __name__ == "__main__":
  main()

File: Codeforces/Round 590 Div. 3/C__AC_bLuck.cpp
#include <bits/stdc++.h>

using namespace std;

typedef long long ll;
typedef double db;
const int mod = 1e9 + 7;

#define INF 1000000000

#define ff first
#define ss second
#define pii pair<int, int>
#define pll pair<ll, ll>
#define sz(a) a.size()
#define all(a) a.begin(), a.end()
#define w(x) cout << #x << " = " << (x) << "\n"

int main() {
#ifndef ONLINE_JUDGE
  freopen("in.txt", "r", stdin);
  // freopen("out.txt", "w", stdout);
#endif
  ios_base::sync_with_stdio(false);
  cin.tie(NULL);
  int t;
  cin >> t;
  while (t--) {
    ll n, k;
    cin >> n;
    string a, b;
    cin >> a >> b;

    if (n == 1) {
      (a[0] <= 2 or b[0] <= 2) ? cout << "NO"
                                      << "\n"
                               : cout << "YES"
                                      << "\n";
      continue;
    }

    bool flag = true, ans = true;
    for (auto i = 0; i < n; i++) {
      if (i != n - 2) {
        if (flag and (a[i] > '2')) {
          if (b[i] <= '2') {
            ans = false;
            break;
          }
          flag = !flag;
        } else if (!flag and (b[i] > '2')) {
          if (a[i] <= '2') {
            ans = false;
            break;
          }
          flag = !flag;
        }
      } else {
        if (flag) {
          if (a[i] <= '2' and (a[i + 1] <= '2')) {
            ans = false;
            break;
          } else if (a[i] > '2' and (b[i] <= '2' or b[i + 1] > '2')) {
            ans = false;
            break;
          }
        } else {
          if (b[i] <= '2' and b[i + 1] > '2') {
            ans = false;
            break;
          } else if (b[i] > '2' and (a[i] <= '2' or a[i + 1] <= '2')) {
            ans = false;
            break;
          }
        }
      }
    }

    (ans) ? cout << "YES"
                 << "\n"
          : cout << "NO"
                 << "\n";
  }
  return 0;
}